### ISSUE TITLE
MIDI out support on Android

### ISSUE THREAD

**Issue by ghost** [openMSX/openMSX#1193]
As the title says, could you please add support for MIDI in the Android version of OpenMSX?
Maybe the easiest way is to use sf2 soundfonts, so the soundbanks can be replaced easily.
http://coolsoft.altervista.org/en/virtualmidisynth#download

Alternative is to support USB-MIDI cable:
https://www.onlinekabelshop.nl/media/catalog/product/cache/1/image/9df78eab33525d08d6e5fb8d27136e95/o/k/oks-87770_hoofd.jpg

Would be nice to have both possibilities.

Can you support this please? Thanks.

**Comment by mthuurne** [comment]
Using SF2 soundfonts would mean integrating a soft synthesizer with MIDI input in openMSX. I'm not sure that's something we should be doing; it smells of feature creep.

There is [MIDI support in the NDK](https://developer.android.com/ndk/reference/group/midi), so we could use that to connect to MIDI devices, either externally such as using the cable you linked, or to a soft synth in a separate app running on the same Android device, assuming Android can do internal MIDI routing like most operating systems can. This would be the approach I would prefer.

For the time being, we have a bigger issue with Android though: we don't have a way yet to build openMSX on Android since the switch to SDL2. Until we solve that, we can't release new Android binaries at all.

**Comment by ghost** [comment]
I agree with you. Your approach would be much nicer. I hope you can build Android versions again in the near future. Anyway, it's cool to use OpenMSX on my Android TV with bluetooth keyboard. It works great.

**Comment by ghost** [comment]
I also found that in the developers options of Android the USB configuration can be set to MIDI. Maybe we can use this as well for OpenMSX? It would be nice to use a PC or other device as soft synth with the use of an USB cable (A to A) besides of an USB to MIDI cable. As far as I know also bluetooth can be used for sending MIDI commands?

**Comment by ghost** [comment]
I close this request for now, because it's already 1,5 year old! But, if MIDI support can be added in the future, that would still be nice.

**Comment by MBilderbeek** [comment]
The request is still valid (as you say so yourself), so there's no need to close it.
